external/VideoStreamer/src/VideoStreamer.cpp

```diff
@@ -0,0 +1,316 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+#include "VideoStreamerInternal.h"
+#include <iostream>
+#pragma comment(lib, "avformat.lib")
+using namespace winrt;
+//#define TIGHT_LATENCY_CONTROL
+#define VERBOSITY  0
+
+#ifdef TIGHT_LATENCY_CONTROL
+    #define MAX_SINK_LATENCY 100
+    uint32_t g_dropCount;
+#endif
+
+#if (VERBOSITY > 0)
+    #define INFOLOG1 printf 
+#else 
+#define INFOLOG1() 
+#endif
+#if (VERBOSITY > 1)
+    #define INFOLOG2 printf 
+#else
+   #define INFOLOG2()
+#endif
+
+    struct GUIDComparer
+{
+    bool operator()(const GUID& Left, const GUID& Right) const
+    {
+        // comparison logic goes here
+        return memcmp(&Left, &Right, sizeof(Right)) < 0;
+    }
+};

```

Comment:
Why does the standard compare not work?

Reply:
Is there a standard way to compare GUID that std::map can use? I didn't see one..

Reply:
you can use ensure you include #include <unknwn.h> since it defines a auto conversion from GUID to winrt::guid
std::map<winrt::guid, winrt::guid>

Reply:
nope. This still does not define a  '<' operator for winrt::guid ,
std::map needs a '<' operator .
"Error C2678 - binary '<': no operator found which takes a left-hand operand of type 'const _Ty' (or there is no acceptable conversion)"